### PR TITLE
TOC markup (fixes #151, fixes #154, fixes #116)

### LINKED ISSUE #116: It would be nice if clicking in the webbook TOC on the small triangle would open the toc with focus on the current chapter
### Prerequisites

_Check the boxes below by changing them from `[ ]` to `[x]`._

* [x] Can you reproduce the behavior you wish to address with [all plugins except Pressbooks deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [x] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/), the latest version of [Pressbooks Book](http://github.com/pressbooks/pressbooks-book/releases/latest/), and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?
* [x] Are you experiencing the issue while using this book theme?
* [x] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fpressbooks-book) to see if your bug or enhancement is already listed?

### Description
It would be nice if clicking in the webbook TOC on the small triangle would open the TOC with focus on the current chapter. See printscreen.

u1

### LINKED ISSUE #151: Optimize TOC and chapter numbering in webbook
### Prerequisites

* [x] Can you reproduce the behavior you wish to address with [all plugins except Pressbooks deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [x] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/), the latest version of [Pressbooks Book](http://github.com/pressbooks/pressbooks-book/releases/latest/), the latest version of [this book theme](http://github.com/pressbooks/pressbooks-book/releases/latest/) and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?
* [x] Are you experiencing the issue while using this book theme?
* [x] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fpressbooks-book) to see if your bug or enhancement is already listed?

### Description

We're still having some consistency/performance issues with part and chapter output in the webbook TOC. To address:

- Part and chapter numbers don't always respect the theme options; sometimes they appear when they shouldn't.
- The `pb_get_chapter_number()` function should be removed wherever possible. One caveat here is that although the book structure array contains all chapters which need to be counted, some of them may be `numberless` which means they need to be skipped in the numbering routine.
- The `pb_get_chapter_number()` function, when used on single chapter views, should be updated to better handle duplicate slugs/avoid collisions, etc.
- TOC markup should include all necessary data/classes/IDs to support https://github.com/pressbooks/pressbooks-book/issues/116; for example, viewing chapter ID 17, in the TOC the relevant chapter should have the ID `chapter-17` so that we can target it with CSS and indicate that it is the current chapter.
- Body classes should include the part ID that contains the current chapter, so that we can expand that part in the TOC dropdown by default.

### Steps to Reproduce

Not applicable.

### System Information

McLuhan: 2.1.0

### LINKED ISSUE #154: Undefined PHP Variables When Looking at URL/table-of-contents/
### Prerequisites

* [X] Can you reproduce the behavior you wish to address with [all plugins except Pressbooks deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [X] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/), the latest version of [Pressbooks Book](http://github.com/pressbooks/pressbooks-book/releases/latest/), the latest version of [this book theme](http://github.com/pressbooks/pressbooks-book/releases/latest/) and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?
* [X] Are you experiencing the issue while using this book theme?
* [X] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fpressbooks-book) to see if your bug or enhancement is already listed?

### Description

`include( locate_template( 'partials/content-toc.php' )` assumes PHP variables are set somewhere higher up the callstack. 

Sometimes they are not.

### Steps to Reproduce

1. Admin -> Organize -> Show In Web -> Uncheck All
2. Delete all transients or change `$output = get_transient( $toc_transient );`  to `$output = false;` in `pressbooks-book/partials/content-toc.php` to duplicate uncached behaviour
2. Go to your book URL/table-of-contents/ -> ie. http://pressbooks.test/test1/table-of-contents/

**Expected behavior:** 
No PHP Warnings in logs

**Actual behavior:** 
PHP Warnings  in logs

### PR DESCRIPTION
**Pull request by connerbw**
Add an id to \<li\> nodes
Add `class=toc__selected` to selected \<li\> node

### CODE REVIEW

File: partials/content-toc.php [hunk 1]
```diff
@@ -46,8 +56,15 @@
...
-	<?php $output = ob_get_clean();
-	echo $output;
-	set_transient( $toc_transient, $output );
+	<?php
+	$toc_output = ob_get_clean();
+	set_transient( $toc_transient, $toc_output );
 }
 
+// Search for [ id="toc-chapter-123" class=" ] and replace (once!) with [ id="toc-chapter-123" class="toc__selected  ]
+if ( isset( $post ) ) {
```
Comment (greatislander) [inline partials/content-toc.php]: This looks great. One other case we need to handle:

```
if ( isset( $post ) ) {}

Comment (greatislander) [inline partials/content-toc.php]: And also:

Comment (connerbw) [reply partials/content-toc.php]: Cool. It already handles part, front-matter, back-matter... The comment is misleading.

Comment (greatislander) [reply partials/content-toc.php]: Right, just to be clear, if I have:

**Pull request merged**